### PR TITLE
Cleanup CHOOSEFONT

### PR DESCRIPTION
**Pull request by hughbe**
## Proposed Changes
- Cleanup CHOOSEFONT

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/2658)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/FontDialog.cs [hunk 1]
```diff
@@ -492,7 +490,7 @@ protected unsafe override bool RunDialog(IntPtr hWndOwner)
             // (limitation of windows control)
 
             Debug.Assert(cf.nSizeMin <= cf.nSizeMax, "min and max font sizes are the wrong way around");
-            if (!SafeNativeMethods.ChooseFont(cf))
+            if (!Comdlg32.ChooseFontW(ref cf).IsTrue())
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/FontDialog.cs]:
```suggestion
            if (Comdlg32.ChooseFontW(ref cf).IsFalse())
```

Comment (hughbe) [reply src/System.Windows.Forms/src/System/Windows/Forms/FontDialog.cs]: I actually stylistically prefer !IsTrue to IsFalse but changed it as requested

File: src/System.Windows.Forms/src/System/Windows/Forms/FontDialog.cs [hunk 2]
```diff
@@ -458,19 +458,17 @@ private void ResetFont()
...
             using ScreenDC dc = ScreenDC.Create();
             using Graphics graphics = Graphics.FromHdcInternal(dc);
-            logFont = User32.LOGFONTW.FromFont(Font, graphics);
+            User32.LOGFONTW logFont = User32.LOGFONTW.FromFont(Font, graphics);
 
-            NativeMethods.CHOOSEFONT cf = new NativeMethods.CHOOSEFONT
+            var cf = new Comdlg32.CHOOSEFONTW
             {
-                lStructSize = Marshal.SizeOf<NativeMethods.CHOOSEFONT>(),
+                lStructSize = (uint)Marshal.SizeOf<Comdlg32.CHOOSEFONTW>(),
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/FontDialog.cs]:
```suggestion
                lStructSize = (uint)sizeof(Comdlg32.CHOOSEFONTW),
```

File: src/System.Windows.Forms/src/System/Windows/Forms/FontDialog.cs (hunk 1 above)
Comment (RussKie) [reply src/System.Windows.Forms/src/System/Windows/Forms/FontDialog.cs]: I agree that in many cases `!IsTrue(....)` may be easier to grasp and comprehend, however I find structures like `!Something(....).IsTrue()` mentally taxing.

File: src/System.Windows.Forms/src/System/Windows/Forms/FontDialog.cs (hunk 2 above)
Comment (hughbe) [reply src/System.Windows.Forms/src/System/Windows/Forms/FontDialog.cs]: > System/Windows/Forms/FontDialog.cs(468,37): error CS0208: Cannot take the address of, get the size of, or declare a pointer to a managed type ('Interop.Comdlg32.CHOOSEFONTW') [/Users/hugh/Documents/GitHub/winforms/src/System.Windows.Forms/src/System.Windows.Forms.csproj]
 Sadly can't

Comment (hughbe) [reply src/System.Windows.Forms/src/System/Windows/Forms/FontDialog.cs]: I think it may be due to the function pointer `User32.WNDPROCINT`

**Pull request merged**